src/Adapter/LinearRegressionAdapter.php

```diff
@@ -2,6 +2,102 @@
 
 namespace Zeeml\Algorithms\Adapter;
 
+use Zeeml\Algorithms\Dataset\DatasetInterface;
+
 class LinearRegressionAdapter extends AbstractAdapter
 {
+    protected $intercept;
+    protected $slope;
+
+    /**
+     * LinearRegressionAdapter constructor.
+     * @param DatasetInterface $dataset
+     */
+    public function __construct(array $data)
+    {
+        $this->trainingData =  $data;
+    }
+
+    /**
+     * prepares the data loaded in order to keep only two dimensional lines and sets keys to numerical
+     * @return $this
+     */
+    public function prepare()
+    {
+        $this->trainingData = array_values(
+            array_filter(
+                $this->trainingData,
+                function(& $line) {
+                    $line = array_values($line);
+                    return count($line) === 2;
+                }
+            )
+        );
+
+        return $this;
+    }
+
+    /**
+     * functions that trains the data sent,
+     * @return bool
+     */
+    public function train()
+    {
+        $this->prepare();
+
+        $numberOfLines = count($this->trainingData);
+
+        //Getting first dimension from each line
+        $x = array_column($this->trainingData, 0);
```

Comment:
shouldn't we have a more generic method to do that? I'm not sure but I seem to remember this process is useful in other algorithms...
e.g. $meanX = $this->colMean(1); // sum values of col 1 and return average.
or  as a static function $meanX = parent::colMean($this->trainingData, 1);